pi/setup_pi.sh

```diff
@@ -39,38 +41,67 @@ else
   # as of Raspbian Buster, this is required for wifi to work
   rfkill unblock wifi
 
+  # install dependencies with apt and pip3
   apt update
   apt install -y darkstat python3-gpiozero autossh ssh parprouted dhcp-helper avahi-daemon python3-pip
   pip3 install python-daemon
 
-  darkstat_init_file=$(<$BASEDIR/darkstat_init.txt)
+  # set up the darkstat config file
+  darkstat_init_file=$(<$BASEDIR/templates/darkstat_init.txt)
   darkstat_init_file="${darkstat_init_file//FLIC_HUB_INTERFACE/$flic_hub_interface}"
   echo "$darkstat_init_file" > /etc/darkstat/init.cfg
+  
+  # set up the dhcp-helper config file
+  # refer to https://wiki.debian.org/BridgeNetworkConnectionsProxyArp
+  dhcp_helper_file=$(<$BASEDIR/templates/dhcp_helper.txt)
+  dhcp_helper_file="${dhcp_helper_file//NETWORK_INTERFACE/$network_interface}"
+  echo "$dhcp_helper_file" > /etc/default/dhcp-helper
+
+  # if we're using wifi then we need to configure wpa_suppplicant
+  wpa_conf_config=''
+  if [[ "$network_interface" == "wlan0"  ]]; then
+    wpa_conf_config='wpa-conf /etc/wpa_supplicant/wpa_supplicant.conf'
+
+    wpa_supplicant_config_file=$(<$BASEDIR/templates/wpa_supplicant.txt)
+    wpa_supplicant_config_file="${wpa_supplicant_config_file//FALLBACK_SSID/$fallback_ssid}"
+    wpa_supplicant_config_file="${wpa_supplicant_config_file//FALLBACK_PSK/$fallback_psk}"
+    wpa_supplicant_config_file="${wpa_supplicant_config_file//SSID/$ssid}"
+    wpa_supplicant_config_file="${wpa_supplicant_config_file//PSK/$psk}"
+    echo "$wpa_supplicant_config_file" > /etc/wpa_supplicant/wpa_supplicant.conf
+  fi
 
-  interfaces_file=$(<$BASEDIR/interfaces.txt)
+  # set up the interfaces file (defines all network interfaces for the system)
+  # refer to https://wiki.debian.org/BridgeNetworkConnectionsProxyArp
+  interfaces_file=$(<$BASEDIR/templates/interfaces.txt)
```

Comment:
I like this move to put the templates into a `/templates` folder. It makes their purpose clearer and the folder more organized.